### PR TITLE
Imp/remark frontmatter defaults

### PR DESCRIPTION
**Pull request by patricksimonian**
add a structure for providing defaults to files when ingested by gatsby

### CODE REVIEW

File: app-web/plugins/gatsby-source-github-all/utils/transformer.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * takes in file parameters and runs it through plugins that
+ * modify the content
+ * @param {String} fileExtension 
+ * @param {String} content 
+ * @param {Object} file 
+ * @returns {String} file content transformed
+ */
+const fileTransformer = (fileExtension, content, file) => ({
+    content,
+    use(plugin, options = {}) {
```
Comment (jleach) [inline app-web/plugins/gatsby-source-github-all/utils/transformer.js]: Where does use come from?

File: app-web/plugins/gatsby-source-github-all/utils/plugins.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        });
+        // create 'new' md string with updated front matter
+        return matter.stringify(data.content, frontmatter);
+    }
+    return raw;
+};
+
+module.exports = {
+    markdownPlugin,
+};
```
Comment (jleach) [inline app-web/plugins/gatsby-source-github-all/utils/plugins.js]: You get these little icons at the bottom of files if they don't have new lines. Generally I think its best practice to have one at the end of a file.

File: app-web/plugins/gatsby-source-github-all/utils/transformer.js (hunk 1 above)
Comment (patricksimonian) [reply app-web/plugins/gatsby-source-github-all/utils/transformer.js]: i think think this is a es6 function definition, similar to 
```javascript
{ 
  content,
  use: function(plugin, options = {}) {...},
}
```

Comment (patricksimonian) [reply app-web/plugins/gatsby-source-github-all/utils/transformer.js]: i may be wrong on its being an es6 definition, it was eslint that actually auto corrected it to what you see now

File: app-web/plugins/gatsby-source-github-all/utils/plugins.js (hunk 2 above)
Comment (patricksimonian) [reply app-web/plugins/gatsby-source-github-all/utils/plugins.js]: gotchya. corrected :)

**Pull request merged**